### ISSUE TITLE
[Bug]: pip install setuptools 67.1.0 issue

### ISSUE THREAD

**Issue by JRowbottomGit** [DLR-RM/stable-baselines3#1324]
### 🐛 Bug

Conda environment with Python version 3.10.9
running: pip install stable-baselines3
**gives error:**
```
Collecting stable-baselines3
  Using cached stable_baselines3-1.7.0-py3-none-any.whl (171 kB)
Collecting gym==0.21
  Using cached gym-0.21.0.tar.gz (1.5 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      error in gym setup command: 'extras_require' must be a dictionary whose values are strings or lists of strings containing valid project/version requirement specifiers.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
```

**The fix is to downgrade setuptools:**
```
pip install -U setuptools==65.5.0

Collecting setuptools==65.5.0
  Using cached setuptools-65.5.0-py3-none-any.whl (1.2 MB)
Installing collected packages: setuptools
  Attempting uninstall: setuptools
    Found existing installation: setuptools 67.1.0
    Uninstalling setuptools-67.1.0:
      Successfully uninstalled setuptools-67.1.0
Successfully installed setuptools-65.5.0
```

### To Reproduce

```python
from stable_baselines3 import ...

```

### Relevant log output / Error message

```shell
Collecting stable-baselines3
  Using cached stable_baselines3-1.7.0-py3-none-any.whl (171 kB)
Collecting gym==0.21
  Using cached gym-0.21.0.tar.gz (1.5 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      error in gym setup command: 'extras_require' must be a dictionary whose values are strings or lists of strings containing valid project/version requirement specifiers.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
```

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a minimal working example to reproduce the bug
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
Duplicate of https://github.com/openai/gym/issues/3176#issuecomment-1408598628

if you need a newer version of setuptools, you can use #780

**Comment by araffin** [comment]
@qgallouedec shall we enforce the version until #780 is ready?

**Comment by qgallouedec** [comment]
> @qgallouedec shall we enforce the version until #780 is ready?

Yes.

**Comment by JRowbottomGit** [comment]
apologies for dupe issue

**Comment by araffin** [comment]
> @qgallouedec shall we enforce the version until https://github.com/DLR-RM/stable-baselines3/pull/780 is ready?

Actually, setuptools is used for build, so I'm not sure that adding it in the requirement will solve the issue.
Anyway, this will be solved by #1327 .

**Comment by ReHoss** [comment]
After running :

`pip3 install setuptools==65.5.0`

and 

`pip install stable-baselines3[extra]`


I get those following errors and warnings (last part of the log output):




```
  ERROR: Failed building wheel for gym
  Running setup.py clean for gym
  Building wheel for lit (setup.py) ... done
  Created wheel for lit: filename=lit-16.0.0-py3-none-any.whl size=93586 sha256=091a629141ac70f758dbdfaf8446fc43dde8273edf422408c885f13fdf640714
  Stored in directory: /gpfs7kw/linkhome/rech/genrqo01/ucd32aq/.cache/pip/wheels/7d/8a/27/f0fa30de83ae4077766dd607b818b68d8b2fac3a6ca649464a
Successfully built lit
Failed to build atari-py gym
Installing collected packages: tensorboard-plugin-wit, pytz, pyasn1, mpmath, lit, cmake, urllib3, tzdata, typing-extensions, tensorboard-data-server, sympy, six, rsa, pyparsing, pyasn1-modules, psutil, protobuf, pillow, packaging, oauthlib, nvidia-nvtx-cu11, nvidia-nccl-cu11, nvidia-cusparse-cu11, nvidia-curand-cu11, nvidia-cufft-cu11, nvidia-cuda-runtime-cu11, nvidia-cuda-nvrtc-cu11, nvidia-cuda-cupti-cu11, nvidia-cublas-cu11, numpy, networkx, MarkupSafe, markdown, kiwisolver, idna, grpcio, fonttools, filelock, cycler, cloudpickle, charset-normalizer, certifi, cachetools, absl-py, werkzeug, requests, python-dateutil, opencv-python, nvidia-cusolver-cu11, nvidia-cudnn-cu11, jinja2, gym, google-auth, contourpy, atari-py, requests-oauthlib, pandas, matplotlib, google-auth-oauthlib, tensorboard, triton, torch, stable-baselines3
  Running setup.py install for gym ... done
  DEPRECATION: gym was installed using the legacy 'setup.py install' method, because a wheel could not be built for it. pip 23.1 will enforce this behaviour change. A possible replacement is to fix the wheel build issue reported above. Discussion can be found at https://github.com/pypa/pip/issues/8368
  Running setup.py install for atari-py ... |^/^_^done
  DEPRECATION: atari-py was installed using the legacy 'setup.py install' method, because a wheel could not be built for it. pip 23.1 will enforce this behaviour change. A possible replacement is to fix the wheel build issue reported above. Discussion can be found at https://github.com/pypa/pip/issues/8368

```


_Preparing metadata (setup.py) _  now worked thanks to downgrading `setuptools`. But I get the following error:


```
 Building wheel for atari-py (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
```

I also have

> 
>       Traceback (most recent call last):
>         File "/tmp/pip-install-ha0wey7k/atari-py_c99e6d0e2b414da6b060a8e040d33a94/setup.py", line 22, in run
>           subprocess.check_call(['cmake', '..'], cwd=cwd)
>         File "/linkhome/rech/genrqo01/ucd32aq/.conda/envs/sb3-contrib/lib/python3.11/subprocess.py", line 408, in check_call
>           retcode = call(*popenargs, **kwargs)
>                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "/linkhome/rech/genrqo01/ucd32aq/.conda/envs/sb3-contrib/lib/python3.11/subprocess.py", line 389, in call
>           with Popen(*popenargs, **kwargs) as p:
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "/linkhome/rech/genrqo01/ucd32aq/.conda/envs/sb3-contrib/lib/python3.11/subprocess.py", line 1024, in __init__
>           self._execute_child(args, executable, preexec_fn, close_fds,
>         File "/linkhome/rech/genrqo01/ucd32aq/.conda/envs/sb3-contrib/lib/python3.11/subprocess.py", line 1901, in _execute_child
>           raise child_exception_type(errno_num, err_msg, err_filename)
>       PermissionError: [Errno 13] Permission denied: 'cmake'
>       
>       During handling of the above exception, another exception occurred:
> 

cmake permission denied. Note that I am not sudo on the computer. So is this comes from the `[extra]` option or from the `atari-py` which is not compatible with gym v21 anymore ?

Thank you:)
